### PR TITLE
Bump go version to 1.19

### PR DESCRIPTION
**Pull request by hbandura**
Upstream cherry pick of go19 build & Dockerfile bump. A godoc whitespace change in go19 provoked many changes in the code.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -20,7 +20,13 @@ linters:
     - misspell
     # - staticcheck
     - unconvert
+    - typecheck
     # - unused
+    - staticcheck
+    - bidichk
+    - durationcheck
+    - exportloopref
+    - whitespace
```
Comment (karlb) [inline .golangci.yml:29]: These cause plenty of [new lint failures](https://github.com/celo-org/celo-blockchain/actions/runs/6285414958/job/17067791540?pr=2189). So either we do not add these checks, or we fix the resulting failures.

**Review by karlb — APPROVED** [review]
Looks good, with the exception of the lint failures.

File: accounts/usbwallet/ledger.go [hunk 2]
```diff
@@ -275,32 +278,32 @@ func (w *ledgerDriver) ledgerVersion() ([3]byte, error) {
...
-//   Public Key length       | 1 byte
-//   Uncompressed Public Key | arbitrary
-//   Celo address length     | 1 byte
-//   Celo address            | 40 bytes hex ascii
-//   Chain code if requested | 32 bytes
+//	Description             | Length
+//	------------------------+-------------------
+//	Public Key length       | 1 byte
+//	Uncompressed Public Key | arbitrary
+//	Ethereum address length | 1 byte
```
Comment (carterqw2) [inline accounts/usbwallet/ledger.go:304]: Is it intentional?

File: build/ci.go [hunk 3]
```diff
@@ -23,21 +24,20 @@ Usage: go run build/ci.go <command> <command flags/arguments>
...
-   aar         [ -local ] [ -sign key-id ] [-deploy repo] [ -upload dest ]                      -- creates an Android archive
-   xcode       [ -local ] [ -sign key-id ] [-deploy repo] [ -upload dest ]                      -- creates an iOS XCode framework
-   xgo         [ -alltools ] [ options ]                                                        -- cross builds according to options
-   xgo-archive [ -targets linux/amd64,linux/386... ][ -type zip|tar ][ -in dir ][ -out dir ]	-- archives build artifacts from cross-compilation
-   purge       [ -store blobstore ] [ -days threshold ]                                         -- purges old archives from the blobstore
+	install    [ -arch architecture ] [ -cc compiler ] [ packages... ]                          -- builds packages and executables
+	test       [ -coverage ] [ packages... ]                                                    -- runs the tests
+	lint                                                                                        -- runs certain pre-selected linters
+	archive    [ -arch architecture ] [ -type zip|tar ] [ -signer key-envvar ] [ -signify key-envvar ] [ -upload dest ] -- archives build artifacts
```
Comment (carterqw2) [inline build/ci.go:30]: Did we support this option?

File: trie/stacktrie.go [hunk 4]
```diff
@@ -370,6 +370,10 @@ func (st *StackTrie) insert(key, value []byte) {
 //   - Then the <32 byte rlp-encoded value will be accessible in 'st.val'.
 //   - And the 'st.type' will be 'hashedNode' AGAIN
 //
+// 2. The rlp-encoded value was < 32 bytes
```
Comment (carterqw2) [inline trie/stacktrie.go:373]: Duplicate.

**Review by carterqw2 — CHANGES_REQUESTED** [review]

File: accounts/usbwallet/ledger.go (hunk 2 above)
Comment (hbandura) [reply accounts/usbwallet/ledger.go:304]: not intentional!

**Review by carterqw2 — APPROVED** [review]

**Pull request merged**